### ISSUE TITLE
Search results can still be improved

### ISSUE THREAD

**Issue by heathermiller** [scalacenter/scaladex#131]
Here's an example:

If I search: `json targets:scala.js_0.6` only a few results on the first page of the search results are meaningful. For example, I see scala-js/scala-js. There's nothing in the README, project description, nor are there any keywords that should cause the scala-js artifacts to be interpreted as a JSON library.

Given this, we should probably restrict our index only on keywords, project name, and project description. 

In general the irrelevant:relevant ratio when searching the index is far too high.

**Comment by olafurpg** [comment]
"shapeless" is another good test query: https://index.scala-lang.org/search?q=shapeless

milessabin/shapeless appears on the 4th page https://index.scala-lang.org/search?q=shapeless&page=4

**Comment by lrytz** [comment]
+1 – i just tried scaladex very superficially so far. the first impression is that search really doesn't yield relevant results first. a typical example: somebody searches for "json", that should give a list of relevant json libs. currently the first page is:

```
propensive/rapture-json-lift
propensive/rapture-json-play
propensive/rapture-json-json4s
propensive/rapture-json-argonaut
propensive/rapture-json-jackson
propensive/rapture-json-spray
propensive/rapture-json-jawn
qifun/json-stream-bson
propensive/rapture-json
thoughtworksinc/microbuilder-core 
```

**Comment by MasseGuillaume** [comment]
fixed in a98347ca4ad493096d016266018fa2ba59468bd0
[shapeless](https://index.scala-lang.org/search?q=shapeless) now on the first page 
[json](https://index.scala-lang.org/search?q=json) shows circe, jawn, json4s on the first page

**Comment by olafurpg** [comment]
The search results have definitely improved, good job 👍

I still feel that the "most relevant" ordering has room for improvements, for queries like "json" I think the star count should weigh more in the ordering even if there exist projects with the exact name "json"

<img width="1118" alt="screen shot 2016-07-27 at 14 41 12" src="https://cloud.githubusercontent.com/assets/1408093/17175598/b22afa52-5408-11e6-9910-9653ca6c72e6.png">


Also, it seems that the search query doesn't match against substrings in repo names. For example, ["test"](https://index.scala-lang.org/search?q=test) doesn't list scalatest. Are descriptions texts stemmed in the search index? Because I think the query "test"  should match projects with "testing" in the description.
